src/transformers/pipelines/fill_mask.py

```diff
@@ -115,25 +115,53 @@ def __call__(self, *args, targets=None, top_k: Optional[int] = None, **kwargs):
         inputs = self._parse_and_tokenize(*args, **kwargs)
         outputs = self._forward(inputs, return_tensors=True)
 
+        # top_k must be defined
+        if top_k is None:
+            top_k = self.top_k
+
         results = []
         batch_size = outputs.shape[0] if self.framework == "tf" else outputs.size(0)
 
         if targets is not None:
-            if len(targets) == 0 or len(targets[0]) == 0:
-                raise ValueError("At least one target must be provided when passed.")
             if isinstance(targets, str):
                 targets = [targets]
 
-            targets_proc = []
+            vocab = self.tokenizer.vocab
+            target_ids = []
             for target in targets:
-                target_enc = self.tokenizer.tokenize(target)
-                if len(target_enc) > 1 or target_enc[0] == self.tokenizer.unk_token:
+                id_ = vocab.get(target, None)
```

Comment:
Here instead of using `vocab.get` which isn't available on all tokenizers, you could encode the target without special tokens and ensure it returns a list with a single value that is not an unknown token.

I can't think of a simpler way that is tokenizer agnostic.

Reply:
If you still want to keep the speed of `vocab.get`, maybe putting it behind a statement such as `if hasattr(self.tokenizer, "vocab")` would help.

Reply:
That's another option, but I don't think we can beat the lookup.

The `tokenizer` part is actually quite slow (compared to a Python dict lookup) here. Especially when called sequentially.
When called batched it was reduced heavily but still non negligible (something like 30ms for 10k targets).
I would have to dive to make sure, but I think it's linked to the strings which have to all be copied and parsed at the Python/Rust interface.

There are other implementations which  I tried that still speedup compared to master, but I had the impression to do anything more than raw lookup for best case scenario was wrong. That's actually what the doc says too, the `targets` should be in the vocab.